pkg/operator/e2e/main_test.go

```diff
@@ -673,63 +673,67 @@ func validateCollectorUpMetrics(ctx context.Context, t *testContext, job string)
 	}
 }
 
-func TestRulesGeneration(t *testing.T) {
-	tctx := newTestContext(t)
+func testRulesGeneration(ctx context.Context, t *testContext) {
+	replace := strings.NewReplacer(
+		"{project_id}", projectID,
+		"{cluster}", cluster,
+		"{namespace}", t.namespace,
+	).Replace
 
 	// Create multiple rules in the cluster and expect their scoped equivalents
 	// to be present in the generated rule file.
-	files := []string{`
+	content := replace(`
```

Comment:
Why use a string literal and `Unmarshal` here over the `ClusterRules` go struct directly? We use this pattern with our other CRDs in the tests `OperatorConfig`, `PodMonitoring`. Same for `Rules`.

Reply:
Yea, I stumbled over this, too. I think some historical indecisiveness :) We should change it at some point.